### PR TITLE
Add minimum software requirements

### LINKED ISSUE #2708: Establish minimum system requirements
### What new functionality do you need?

As the number of supported systems increases, minimum requirements to run the workflow should be established.  This includes spack-stack, bash, and Python and possibly lmod, and slurm/pbspro.

### What are the requirements for the new functionality?

All minimum versions have been tested on at least one supported platform.

### Acceptance Criteria

Documentation is updated accordingly.

### Suggest a solution (optional)

Bash: 4.4.20+
Python: 3.8+
spack-stack: 1.6.0
lmod: 8.3.1
slurm: 22+
PBSpro: 2022.1.1+
git: 2.18

### PR DESCRIPTION
**Pull request by DavidHuber-NOAA**
# Description
Adds a table to HPC documentation stating the minimum support versions.

Resolves #2708

# Type of change
- Documentation

# Change characteristics
- Does this change require a documentation update? YES

# How has this been tested?
Render the docs in this PR

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: docs/source/hpc.rst [hunk 1]
```diff
@@ -4,6 +4,44 @@ HPC Settings and Help
...
++--------------+-------------+---------------------------------------+
+| Spack-Stack  | 1.6.0       | * Available everywhere but WCOSS2     |
++--------------+-------------+---------------------------------------+
+| lmod         | 8.3.1       |                                       |
++--------------+-------------+---------------------------------------+
+| Slurm        | 23.02.7     | * Other schedulers may be supportable |
++--------------+-------------+---------------------------------------+
+| PBSpro       | 2022.1.1    | * Other schedulers may be supportable |
++--------------+-------------+---------------------------------------+
+| Git          | 2.29.0      |                                       |
```
Comment (aerorahul) [inline docs/source/hpc.rst]:
```suggestion
| Git          | 2.29.0      |                                       | * Some components e.g. GDASApp may need Git-LFS for downloading test data
```
Just dropping a note on Git-LFS used in some of the test data of GDASApp.
It is not a requirement for global-workflow.

File: modulefiles/module_gwsetup.gaea.lua [hunk 2]
```diff
@@ -15,7 +15,5 @@ load(pathJoin("python", python_ver))
 load("py-jinja2")
 load("py-pyyaml")
 load("py-numpy")
-local git_ver=os.getenv("git_ver") or "2.35.2"
```
Comment (aerorahul) [inline modulefiles/module_gwsetup.gaea.lua:18]: Are you sure about this?  
I think the ci scripts are using git, and just making sure this is not going to blow up there.

Comment (DavidHuber-NOAA) [reply modulefiles/module_gwsetup.gaea.lua:18]: Yes, I tested the system `git` version on all platforms.  The only system that needs a non-default version is S4.  Just to be safe, I can revert the changes here and add this to a future PR that needs to be CI tested.  Thoughts?

Comment (aerorahul) [reply modulefiles/module_gwsetup.gaea.lua:18]: No.  This is good enough for me.  I was just double-checking.

**Review by aerorahul — APPROVED** [review]
looks good.
Thanks!

**Pull request merged**